dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -469,6 +469,46 @@ public PhaseOneResult process(CompilationUnitTree root, UnderlyingAST underlying
    */
   public void handleArtificialTree(Tree tree) {}
 
+  @Override
+  public Node scan(Tree tree, Void p) {
+    if (tree == null) {
+      return null;
+    }
+    // Must use String comparison to support compiling on JDK 11 and earlier.
+    if (tree.getKind().name().equals("SWITCH_EXPRESSION")) {
+      return visitSwitchExpression17(tree, p);
+    }
+    return super.scan(tree, p);
+
+    // TODO: Do we need to support yield trees and binding patterns to?
+    // Features added between JDK 12 and JDK 17 inclusive.
+    // switch (tree.getKind().name()) {
+    //   case "BINDING_PATTERN":
+    //     return visitBindingPattern17(path.getLeaf(), p);
+    //   case "SWITCH_EXPRESSION":
+    //     return visitSwitchExpression17(tree, p);
+    //   case "YIELD":
+    //     return visitYield17(path.getLeaf(), p);
+    //   default:
+    //     return super.scan(tree, p);
+    // }
+  }
+
+  /**
+   * Visit a SwitchExpressionTree
+   *
+   * @param switchExpressionTree a SwitchExpressionTree, typed as Tree to be backward-compatible
+   * @param p parameter
+   * @return the result of visiting the switch expression tree
+   */
+  public Node visitSwitchExpression17(Tree switchExpressionTree, Void p) {
+    // TODO: Analyze switch expressions properly.
+    return new MarkerNode(
```

Comment:
I decided not to add a SwitchExpressionNode, because I wasn't sure what fields it would need when we properly implement switch expressions.